prow/cmd/exporter/main.go

```diff
@@ -54,16 +55,13 @@ func (o *options) Validate() error {
 	return o.kubernetes.Validate(false)
 }
 
-func mustRegister(component string, lister lister) *prometheus.Registry {
```

Comment:
Wait so what happened to the original intent of this method? Why did you remove it?

Reply:
If the original idea was to not export the metrics of the exporter to the push-gateway then it was fine. The `registry` parameter which you can pass to the `push` function is only exposed at `/metrics` and not pushed. I re-wrote it do look like it does in the other components. There the go instrumentation is already registered via `init`.

Reply:
The intent was to create a registry and expose that via a handler in the server for metrics. You've removed that and always pass `nil`. Why? The label metrics are complex, let's revert this change unless you run this component in your cluster and can verify at runtime that `prow_job_labels` and `prow_job_annotations` work as expected.

Reply:
Feel free to register the new histograms using the native client methods in an `init()`

Reply:
Let's change absolutely nothing about the way they are handled. They are circumventing the canonical Go client usecase on purpose.

Reply:
Ack, Changed.